### ISSUE TITLE
[Question] Used for OKD?

### ISSUE THREAD

**Issue by Alexr03** [RedHat-EMEA-SSA-Team/hetzner-ocp4#136]
Is it possible that this could instead be used to install/use OKD 4.5?

**Comment by sandrich** [comment]
I have used it to install OKD but had to do some modifications

**Comment by rbo** [comment]
> I have used it to install OKD but had to do some modifications. If I find time I will upload my diff to my forked repository.

Very interesting, I would like to try to run OKD as a well. Please share your changes. Maybe we can adjust the code a bit for easy OKD usage.

**Comment by sandrich** [comment]
Here a raw git diff. Sorry for the raw material. It was pretty straight forward. OKD provides a different image compression. So had to work around that a bit.

```
diff --git a/ansible/roles/openshift-4-cluster/tasks/create-vm.yml b/ansible/roles/openshift-4-cluster/tasks/create-vm.yml
index 44670cc..313b4ce 100644
--- a/ansible/roles/openshift-4-cluster/tasks/create-vm.yml
+++ b/ansible/roles/openshift-4-cluster/tasks/create-vm.yml
@@ -7,12 +7,16 @@
   args:
     creates: "/var/lib/libvirt/images/{{ vm_instance_name }}.qcow2"
 
+#- name: Unarchive      
+#  copy:
+#    src: "{{ coreos_image_location }}"
+#    dest: "/var/lib/libvirt/images/{{ vm_instance_name }}.qcow2"
+
 - name: Copy ignition {{ vm_instance_name }}
   copy:
     src: "{{ vm_ignition_file }}"
     dest: "/var/lib/libvirt/images/{{ vm_instance_name }}.ign"
 
-# - name: Debug - create /tmp/{{ vm_instance_name }}.virt.xml
 #   template:
 #     src: "vm.xml.j2"
 #     dest: "/tmp/{{ vm_instance_name }}.virt.xml"
diff --git a/ansible/roles/openshift-4-cluster/tasks/download-openshift-artifacts.yml b/ansible/roles/openshift-4-cluster/tasks/download-openshift-artifacts.yml
index b9619dc..279e7b2 100644
--- a/ansible/roles/openshift-4-cluster/tasks/download-openshift-artifacts.yml
+++ b/ansible/roles/openshift-4-cluster/tasks/download-openshift-artifacts.yml
@@ -2,8 +2,10 @@
 
 - name: Build openshift download urls
   set_fact:
-    tmp_openshift_install_download_url: "{{ openshift_location }}/openshift-install-linux-{{ openshift_version }}.tar.gz"
-    tmp_openshift_client_download_url: "{{ openshift_location }}/openshift-client-linux-{{ openshift_version }}.tar.gz"
+#    tmp_openshift_install_download_url: "{{ openshift_location }}/openshift-install-linux-{{ openshift_version }}.tar.gz"
+#    tmp_openshift_client_download_url: "{{ openshift_location }}/openshift-client-linux-{{ openshift_version }}.tar.gz"
+    tmp_openshift_install_download_url: https://github.com/openshift/okd/releases/download/4.4.0-0.okd-2020-04-21-163702-beta4/openshift-install-linux-4.4.0-0.okd-2020-04-21-163702-beta4.tar.gz
+    tmp_openshift_client_download_url: https://github.com/openshift/okd/releases/download/4.4.0-0.okd-2020-04-21-163702-beta4/openshift-client-linux-4.4.0-0.okd-2020-04-21-163702-beta4.tar.gz
     
 
 - name: Check download urls
@@ -13,8 +15,8 @@
     status_code: 200
   with_items:
     - "{{ coreos_download_url }}"
-    - "{{ tmp_openshift_install_download_url }}"
-    - "{{ tmp_openshift_client_download_url }}"
+#    - "{{ tmp_openshift_install_download_url }}"
+#    - "{{ tmp_openshift_client_download_url }}"
     - "{{ helm_cli_location }}"
 
 # Can not use get_url because: get_url do not support --compressed
@@ -30,6 +32,9 @@
     path: "{{ coreos_image_location }}"
   register: coreos_image
 
+- debug:
+    var: coreos_image
+
 - block:
   - name: Rename image -> image.gzip
     command: "mv {{ coreos_image_location }} {{ coreos_image_location }}.gz"
@@ -37,6 +42,13 @@
     command: "gzip -d {{ coreos_image_location }}.gz"
   when: coreos_image.stat.mimetype == "application/x-gzip"
 
+- block:
+  - name: Rename
+    command: "mv {{ coreos_image_location }} {{ coreos_image_location }}.xz"
+  - name: Uncompress
+    command: "unxz {{ coreos_image_location }}.xz"
+  when: coreos_image.stat.mimetype == "application/x-xz"
+
 - name: CoreOS image checksum check
   stat:
     path: "{{ coreos_image_location }}"
@@ -60,19 +72,19 @@
   - install
   - client
 
-- name: Download Openshift installer
-  unarchive:
-    src: "{{ tmp_openshift_install_download_url }}"
-    dest: "/opt/openshift-install-{{ openshift_version }}/"
-    remote_src: yes
-    creates: "/opt/openshift-install-{{ openshift_version }}/openshift-install"
-
-- name: Download Openshift client
-  unarchive:
-    src: "{{ tmp_openshift_client_download_url }}"
-    dest: "/opt/openshift-client-{{ openshift_version }}/"
-    remote_src: yes
-    creates: "/opt/openshift-client-{{ openshift_version }}/oc"
+#- name: Download Openshift installer
+#  unarchive:
+#    src: "{{ tmp_openshift_install_download_url }}"
+#    dest: "/opt/openshift-install-{{ openshift_version }}/"
+#    remote_src: yes
+#    creates: "/opt/openshift-install-{{ openshift_version }}/openshift-install"
+#
+#- name: Download Openshift client
+#  unarchive:
+#    src: "{{ tmp_openshift_client_download_url }}"
+#    dest: "/opt/openshift-client-{{ openshift_version }}/"
+#    remote_src: yes
+#    creates: "/opt/openshift-client-{{ openshift_version }}/oc"
 
 - name: Download Helm client
   get_url:
diff --git a/ansible/roles/openshift-4-cluster/tasks/prepare-host-RedHat-8.yml b/ansible/roles/openshift-4-cluster/tasks/prepare-host-RedHat-8.yml
index b902df5..6020a24 100644
--- a/ansible/roles/openshift-4-cluster/tasks/prepare-host-RedHat-8.yml
+++ b/ansible/roles/openshift-4-cluster/tasks/prepare-host-RedHat-8.yml
@@ -9,13 +9,8 @@
       # ansible virt need lxml
       - python3-lxml
       - python3-openshift
-      - python3-boto
     state: present
 
-- name: Upgrade all packages
-  yum:
-    name: '*'
-    state: latest
 
 - name: Allow NFS traffic from VM's to Host
   firewalld:
@@ -55,4 +50,4 @@
   notify: 'reload firewalld'
 
 - name: firewalld reload
-  command: firewall-cmd --reload
\ No newline at end of file
+  command: firewall-cmd --reload
```

**Comment by rbo** [comment]
@sandrich thanks.

What is your current head? `git rev-parse HEAD` 

I can not apply the patch:
```
git apply diff.patch
diff.patch:9: trailing whitespace.
#- name: Unarchive      
error: patch failed: ansible/roles/openshift-4-cluster/tasks/create-vm.yml:7
error: ansible/roles/openshift-4-cluster/tasks/create-vm.yml: patch does not apply
error: patch failed: ansible/roles/openshift-4-cluster/tasks/download-openshift-artifacts.yml:2
error: ansible/roles/openshift-4-cluster/tasks/download-openshift-artifacts.yml: patch does not apply
```

**Comment by sandrich** [comment]
@rbo pretty old one ```2fda0a969344d9ff64ff5e3ebd3fb7751f01cb6a```

**Comment by Alexr03** [comment]
@sandrich This is great! Thanks for sharing

**Comment by Alexr03** [comment]
After cloning and doing:
`git checkout 2fda0a969344d9ff64ff5e3ebd3fb7751f01cb6a` then attempting to patch it, I am getting 

```
error: patch failed: ansible/roles/openshift-4-cluster/tasks/download-openshift-artifacts.yml:2
error: ansible/roles/openshift-4-cluster/tasks/download-openshift-artifacts.yml: patch does not apply
```

**Comment by sandrich** [comment]
I will pull the latest commit and implement it properly tomorrow. Unfortunately I don't have a spare server to test but will let you know

**Comment by rbo** [comment]
I created a branch called okd4 with the patch based on old head. 

@sandrich please use the devel branch.

**Comment by Alexr03** [comment]
No worries, I have a server I can test on 👌

**Comment by rbo** [comment]
Merged into okd4 branch to run further tests.

**Comment by rbo** [comment]
@sandrich thanks for your contribution. Merged into devel after some small changes. Will be merged into master with PR  #126
